packages/CodeQuality/tests/Rector/For_/ForRepeatedCountToOwnVariableRector/Fixture/fallback_for_complex.php.inc

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Rector\CodeQuality\Tests\Rector\For_\ForRepeatedCountToOwnVariableRector\Fixture;
+
+class FallbackForComplex
+{
+    public function run($items, \stdClass $someObject)
+    {
+        for ($i = 5; $i <= count($someObject->getItems() + 10); $i++) {
+            echo $items[$i];
```

Comment:
In case the loop body changes the number of elements of the array the `count()` cannot be calculated beforehand .. could happen when the array get re-assigned, changed by reference or elements get added/deleted

Reply:
There are many theoretical edge cases. Not worth covering unless really exist